cmd/docker/main.go

```diff
@@ -130,16 +136,31 @@ func handleFlags() bool {
 	}
 	extraconfig.Decode(src, &vchConfig)
 
-	if *cli.debug || vchConfig.Diagnostics.DebugLevel > 0 {
-		log.SetLevel(log.DebugLevel)
-	}
-
 	*cli.portLayerAddr = fmt.Sprintf("%s:%d", *cli.portLayerAddr, *cli.portLayerPort)
 	cli.proto = "tcp"
 
 	return true
 }
 
+func initLogging() error {
+	logcfg := viclog.NewLoggingConfig()
+	if *cli.debug || vchConfig.Diagnostics.DebugLevel > 0 {
+		logcfg.Level = log.DebugLevel
+	}
+
+	if vchConfig.Diagnostics.SysLogConfig != nil {
+		logcfg.Syslog = &syslog.SyslogConfig{
+			Network:   vchConfig.Diagnostics.SysLogConfig.Network,
```

Comment:
We never completed #599 so there's still some misalignment. We also need to consider how read-only configuration such as this could and should be exposed for regular containers in order to be consistent with the vic-machine/pl work.